### PR TITLE
Add unitary update

### PR DESCRIPTION
**Pull request by sylviemonet**
Fix the bugs about the Riemannian optimization.

Add two optimization methods of the Interferometer (unitary update) and RealInterferometer (orthogonal update).

### CODE REVIEW

File: mrmustard/training/parameter.py [hunk 1]
```diff
@@ -174,17 +178,27 @@ def __init__(self, value: Any, name: str, owner: Optional[str] = None) -> None:
...
     def __init__(self, value: Any, name: str, owner: Optional[str] = None) -> None:
-        self._value = (
-            value
-            if math.from_backend(value) and not math.is_trainable(value)
-            else math.new_constant(value, name)
-        )
+        if math.from_backend(value) and math.is_trainable(value):
+            self._value = value
+        elif type(value) in [list, int, float]:
+            self._value = math.new_variable(value, bounds=None, name=name)
```
Comment (ziofil) [inline mrmustard/training/parameter.py:199]: this is the `Constant` class, why are we making a variable?

**Review by ziofil — COMMENTED** [review]
Nice! I only have that comment on the Constant class (which I didn't notice earlier)

File: mrmustard/training/parameter.py (hunk 1 above)
Comment (sylviemonet) [reply mrmustard/training/parameter.py]: You are right. I forgot why I changed here. I just use the previous version of this part.

**Review by ziofil — APPROVED** [review]
looks good! :)

**Pull request merged**